tools/interop_matrix/create_matrix_images.py

```diff
@@ -214,6 +214,24 @@ def cleanup():
 docker_images_cleanup = []
 atexit.register(cleanup)
 
+def maybe_perform_hacks_on_git_repo(stack_base, lang, release):
+  if lang != 'ruby' or release != 'v1.0.1':
```

Comment:
I think ruby won't be the only language that will ever need this (C# v1.0.1 might be the next one), so we should make the patching work in a more organized way.
Also, any patches to the dockerfiles we applied so far (to be able to build images for older releases) should really be persisted in the repo in the same way - I don't think manually hacking the dockerfiles locally and then pushing the images is something we should do.

Here's what I'd suggest:
the client_matrix.py should contain info about what patches to apply for given versions and the patching code here should be general and apply patches according to that list.

Reply:
SG. 

My idea for persisting the hacks in our git repo was really just the presence of those patch files, and the way they are applied in the script. The uploaded images will have the `commit_log` file with the "hack commit" appended to it with it's timestamp, as well as the whole git repo with that hack commit, which IMO might help.

The only other thing I can think of as a way to store these hacks in version control is to store the <i>results</i> of the patches, i.e. store the whole patched docker file somewhere.

What do you think?

Reply:
Excellent advice, Jan!

Reply:
I think patches might be better than storing the entire files:
- you can patch more than one file with a single .patch file
- it makes the changes more obvious
- (potentially) patches can be stacked on top of each other